tools/slicec-cs/src/encoding.rs

```diff
@@ -351,9 +355,10 @@ fn encode_action_body(
     let mut code = CodeBlock::default();
     let is_optional = type_ref.is_optional && !is_tagged;
 
-    let value = match (is_optional, type_ref.is_value_type()) {
-        (true, false) => "value!",
-        (true, true) => "value!.Value",
+    let value = match (is_optional, type_ref.is_value_type(), type_ref.is_reference_type()) {
+        (true, false, true) => "value!",
+        (true, true, false) => "value!.Value",
+        (true, false, false) => "(value ?? default!)",
```

Comment:
I would explicitly mark this case as unreachable, since it's impossible for something to be both a value and reference type.
```suggestion
        (true, false, false) => "(value ?? default!)",
        (true, true, true) => unreachable!(),
```

Reply:
It is possible for custom, isn't it? it just mean we don't know

Reply:
The functions return true if we know-for-sure this is mapped to a value/reference type.
So custom is handled by `(_, false, false)`
Since we don't know that it's value type and we also don't know that it's a reference type.

`(_, true, true)` would mean we know something is a value type, and we also know it's a reference type, which is impossible.